[BUG] Main build broken — undefined symbols + voiceEvent signature drift

## 🎯 Cosa stiamo facendo (context)
Il main è broken da `df5a645` (saga wake-word/LiveActivity/orchestrator). Il commit ha aggiunto chiamate a simboli mai definiti + cambiato la signature di `voiceEvent` senza aggiornare i call site. 20 commit dopo nessuno fixa. Blocca **#9** + ogni iOS dev con build verify.

## 🔧 Cosa implementeremo
- Rimuovere label `turnId:` dalla signature `voiceEvent` (tutti i 10 call site sono positional)
- Definire `ensureVoiceTurn(reason:) -> String` e `clearPendingDone(reason:)` come private helper in `GigiSmartOrchestrator` (semantica recuperata dai call site)
- Build verify su PeterPan_Iphone

## ✨ Risultato atteso (cosa cambia per il team)
Main builda di nuovo. Sblocca #9, sblocca Fede, sblocca ogni iOS task della settimana.

---

## File toccati attesi
- `02_GIGI_APP/GIGI/GigiDebugLogger.swift` — signature `voiceEvent` (1 riga)
- `02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift` — 2 nuovi private method

## Errori build osservati (BUILD FAILED)
```
GigiSmartOrchestrator.swift:163: cannot find 'ensureVoiceTurn' in scope
GigiSmartOrchestrator.swift:352: cannot find 'ensureVoiceTurn' in scope
GigiSmartOrchestrator.swift:373: cannot find 'ensureVoiceTurn' in scope
GigiSmartOrchestrator.swift:384: cannot find 'ensureVoiceTurn' in scope
GigiSmartOrchestrator.swift:385: cannot find 'clearPendingDone' in scope
GigiSmartOrchestrator.swift:157,251,265,274: cannot convert 'String?' to '[String : String]'
GigiSmartOrchestrator.swift:164,274,386: missing argument labels 'turnId:location:function:line:'
GigiAudioManager.swift:158,190,197,214,244: same voiceEvent signature errors
```

## Root cause
- `df5a645` ha introdotto 5 call site di `ensureVoiceTurn`/`clearPendingDone` ma `git log -S` su tutti i branch conferma che le definizioni **non sono mai esistite** → resolution conflict mancata o file dimenticato fuori commit.
- Nello stesso commit `voiceEvent` è stata cambiata da `(_, _, _)` a `(_, turnId:, _)` ma nessun call site usa la label.

## Acceptance criteria
- [ ] `xcodebuild build -scheme GIGI -destination 'id=00008110-001608EC02F0A01E'` → BUILD SUCCEEDED
- [ ] App si avvia su iPhone fisico (PeterPan_Iphone) senza crash al cold launch
- [ ] Wake word "hey gigi" → app passa a listening (senza regressione di stato turno)
- [ ] No nuovi warning/error in `GigiDebugLogger`, `GigiSmartOrchestrator`, `GigiAudioManager`

## Branch
`fix/main-build-broken` (worktree off main)

cc @ArmandoBattaglino — sblocca #9 stand-by + tutti gli iOS task settimana lancio.

--- a/02_GIGI_APP/GIGI/GigiDebugLogger.swift
+++ b/02_GIGI_APP/GIGI/GigiDebugLogger.swift
@@ -29,7 +29,7 @@ class GigiDebugLogger {
 
     static func voiceEvent(
         _ name: String,
-        turnId: String? = nil,
+        _ turnId: String? = nil,
         _ data: [String: String] = [:],
         location: String = #file,
         function: String = #function,

--- a/02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift
+++ b/02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift
@@ -414,6 +414,29 @@ class GigiSmartOrchestrator: ObservableObject {
         Task { await GigiLiveActivityController.shared.endImmediately() }
     }
 
+    // MARK: - Voice turn lifecycle
+
+    // Idempotent: callers (transcript, quickTalk, wakeOrTap, interrupt) just want a
+    // non-nil turnId without caring whether one is already in flight. A barge-in keeps
+    // the existing turn so structured logs (`orchestrator.interruptAndListen`) continue
+    // the same trace; first call after `fireDone` clears `currentVoiceTurnId` allocates
+    // a fresh one.
+    @discardableResult
+    private func ensureVoiceTurn(reason: String) -> String {
+        if let existing = currentVoiceTurnId { return existing }
+        let id = UUID().uuidString.prefix(8).description
+        currentVoiceTurnId = id
+        GigiDebugLogger.voiceEvent("orchestrator.turnStart", id, ["reason": reason])
+        return id
+    }
+
+    private func clearPendingDone(reason: String) {
+        GigiDebugLogger.voiceEvent("orchestrator.clearPendingDone", currentVoiceTurnId, ["reason": reason])
+        pendingDoneMessage = nil
+        doneSafetyTask?.cancel()
+        doneSafetyTask = nil
+    }
+
     // MARK: - Helpers
 
     /// Splits a text containing multiple sequential commands into individual parts.

